Post-launch Additions and Edits

jhogue opened the pull request:
A few things that I left hanging without a pull request. Getting back into it. 

* Add touch icons
* Use the icomoon font instead of FA for the footer social icons
* Adjust some colors and padding
* Add a conditional around the banner image

jhogue committed:
Add touch icons and associated meta tags

Made with realfavicongenerator.net

jhogue committed:
Conditionally hide the Banner, adjust colors

Add an is_front_page() check. Adjust colors in the banner and make the blue a little lighter, while adding a variable for an additional blue for border (the original blue shade)

jhogue committed:
Header design adjustments

jhogue committed:
Add Instagram to footer, use built-in icomoon

The theme has icomoon, and yet the code for the footer was using FontAwesome. Removed the code that called fontawesome icons and instead used icomoon.

jhogue committed:
Add focus states to hovers, adjust padding

Padding needed to be increased for the two color blocks that contain the Who is PPS  and the How You Can Help stories.

kangabell commented on header.php:
```diff
@@ -23,6 +23,31 @@
+  <meta name="application-name" content="PPSri.org">
+  <meta name="msapplication-TileColor" content="#4881cb">
+  <meta name="msapplication-TileImage" content="<?php echo get_stylesheet_directory_uri() ?>/library/icons/mstile-150x150.png">
+  <meta name="msapplication-TileImage" content="<?php echo get_stylesheet_directory_uri() ?>/library/icons/mstile-144x144.png">
```
Why not use wordpress core's Site Icon? https://www.sitepoint.com/all-you-need-to-know-about-the-new-wordpress-site-icon-api/ You just upload an image once and it will create much of this for you. You can set up custom sizes for anything you want that isn't there by default.

jhogue replied:
WordPress core favicon is incomplete. Or, it covers the very basic subset of icons needed for mobile and desktop. When I use WP core it only produces four lines of code, covering iOS and classic desktop browsers. Android, Windows, & OSX El Capitan are not covered and the checker at realfavicongenerator.net says that the `favicon/ico` file is ill-formed and missing some of the recommended embedded sizes. The tile icon that is created for Metro is just the same as the touch icon, therefore it does not adhere to the Windows Metro standards of a white transparent PNG. 

On top of all that, I have curated the different sizes needed with different artwork. As the icon gets smaller, the complicated logo artwork becomes cropped and simpler so there is still some legibility of the mark even at the 16x16 size. 

In short, I prefer to complete set of meta data rather than WP's minimum.

kangabell replied:
Okay! Thanks for the explanation. I think there is a way to hook into the existing core functionality and expand it, but I'm not sure if you can "curate" the sizes like you're saying. Let's go with what you have.

kangabell commented on header.php:
```diff
@@ -23,6 +23,31 @@
+  <meta name="theme-color" content="#d1dcf2">
+
+<?php if ( ! is_user_logged_in() ) { ?>
+  <!-- Global site tag (gtag.js) - Google Analytics -->
```
trackers make me sad : ( but I understand clients want them.

jhogue replied:
For the record, the only tracker they use is the standard Analytics, but GA wants you to use their global site tag now as a way of managing it. I suppose I could go around it and only install the standard GA… but I dont know if PPS will want to use another service like AdWords in the future.

kangabell commented on library/icons/html_code.html:
```diff
@@ -0,0 +1,11 @@
+<meta name="application-name" content="PPS.org">
+<meta name="msapplication-TileColor" content="#4881cb">
+<meta name="msapplication-config" content="/wp-content/themes/pps-2017/browserconfig.xml">
+<meta name="theme-color" content="#d1dcf2">
```
I think this is the third time we've seen this chunk of code?

jhogue replied:
What code exactly?

jhogue replied:
Oh oh oh… I could remove this file completely, actually. This is the same code that I already put into header.php. Its just a copy of the code that comes with the download package of icons. Sort of like when you download Icomoon and it included the sample CSS/HTML.

kangabell commented on library/sass/_setup.scss:
```diff
@@ -68,13 +69,8 @@ $font-weight-bold: 900;
-    border-bottom: 6px solid;
-  }
+@mixin border($direction:null,$color:$blue-border) {
+  border-#{$direction}: 5px solid;
```
I think this line needs an `@if`, because `$direction` is initially set to `null`. if in our usage we don't specify `$direction` (e.g. `@include border();`) it would break right?

jhogue replied:
Correct, good catch. Instead of an @if, I will add a `!default` value of `top`. Still good to pass a value of your own through, even if it is `top` for readability when used.

jhogue replied:
For even better mixin-ness, I could get it to accept a string of values for border direction (like `top left`), and then split the string on a space and feed it to a @each loop… but you seem to only use this when setting one border, so that seems like overkill.

kangabell replied:
`!default` is a good idea that I hadn't thought of! and yeah I agree that expanding the mixin for different border directions is interesting but seems like overkill.

kangabell commented:
Some notes! My main concern is with this favicon stuff. It seems like a lot of code to duplicate something that's already available in core.

jhogue committed:
Don’t let a value pass as null

Null would produce ill-formed CSS — i.e. `border-: 5px solid`. It would work if the hyphen was’t there.

kangabell commented:
@jhogue Wanna just remove that extra file you mentioned?

jhogue committed:
Remove doc files from external sources
